controllers/validation.js

```diff
@@ -0,0 +1,626 @@
+		default:
+			generateFileErrors(output, error, messages);
+			break;
+		}
+	});
+	errorMessage = concatErrors(messages);
+	return errorMessage;
+
+}
+
+function checkForFilesInSchema(schema, toCheck){
+	
+	const keys = Object.keys(schema);
+	keys.forEach((key)=>{
+		switch (key){
+		case 'allOf':
+			schema.allOf.forEach((sch)=>{
+				checkForFilesInSchema(sch, toCheck);
+			});
+			break;
+		case 'properties':
+			checkForFilesInSchema(schema.properties, toCheck);
+			break;
+		default:
+			if (schema[key].type === 'file'){
+				const obj = {};
+				obj[key] = schema[key];
+				toCheck.push(obj);
+			}
+			else if (schema[key].type === 'object'){
+				checkForFilesInSchema(schema[key], toCheck);
+			}
+			break;
+		}
+	});
+}
+
+function getFileInfo(file, constraints){
+	const uploadFile = {};
+	const uploadField = Object.keys(constraints)[0];
+	if (file){
+		const filename = path.parse(file[0].originalname).name;
+
+		uploadFile.file = file[0];
+		uploadFile.originalname = uploadFile.file.originalname;
+		uploadFile.filetype = Object.keys(constraints)[0];
+		uploadFile.filetypecode = constraints[uploadFile.filetype].filetypecode;
+		uploadFile.ext = path.parse(uploadFile.file.originalname).ext.split('.')[1];
+		uploadFile.size = uploadFile.file.size;
+		uploadFile.mimetype = uploadFile.file.mimetype;
+		uploadFile.encoding = uploadFile.file.encoding;
+		uploadFile.buffer = uploadFile.file.buffer;
+		uploadFile.filename = uploadField + '-' + filename + '-' + Date.now() + '.' + uploadFile.ext;
+
+	}
+
+	return uploadFile;
+}
+
+function validateFile(uploadFile, validationConstraints, fileName){
+
+	const fileInfo = getFileInfo(uploadFile, validationConstraints);
+	const constraints = validationConstraints[fileName];
+	const regex = `(^${constraints.validExtensions.join('$|^')}$)`;
+	const errObjs = [];
+
+	if (uploadFile){
+		if (fileInfo.ext && !fileInfo.ext.toLowerCase().match(regex)){
+			errObjs.push(makeErrorObj(fileInfo.filetype, 'invalidExtension', constraints.validExtensions));
+		}
+		else if (fileMimes.indexOf(fileInfo.mimetype) < 0){
+			errObjs.push(makeErrorObj(fileInfo.filetype, 'invalidMime', fileMimes));
+		}
+		if (fileInfo.size === 0){
+			errObjs.push(makeErrorObj(fileInfo.filetype, 'invalidSizeSmall', 0));
+		}
+		else {
+			const fileSizeInMegabytes = fileInfo.size / 1000000.0;
+			if (fileSizeInMegabytes > constraints.maxSize){
+				errObjs.push(makeErrorObj(fileInfo.filetype, 'invalidSizeLarge', constraints.maxSize));
+			}
+		}
+	}
+	else if (constraints.requiredFile){
+		errObjs.push(makeErrorObj(fileName, 'requiredFileMissing'));
+	}
+
+	return errObjs;
+	
+}
+
+function getFieldValidationErrors(body, pathData, derefSchema){
+	const processedFieldErrors = {
+		errorArray:[]
+	};
+	const fieldErrors = validateBody(body, pathData);
+	if (fieldErrors.length > 0){
+		processErrors(fieldErrors, processedFieldErrors, derefSchema);
+	}
+
+	return processedFieldErrors;
+}
+
+module.exports.removeInstance = removeInstance;
```

Comment:
Probably don't need to expose all functions. Should only expose those that are used outside the function..